addon_service/common/hmac.py

```diff
@@ -1,62 +1,49 @@
 import base64
+import functools
 import hashlib
 import hmac
 import re
-import urllib
+import urllib.parse
 from datetime import (
     UTC,
     datetime,
+    timedelta,
 )
 
-from django.conf import settings
+from django.http import HttpRequest
 
 
+__all__ = (
+    "make_signed_headers",
+    "validate_signed_headers",
+    "TIMESTAMP_HEADER",
+)
+
+_AUTH_HEADER_PREFIX = "HMAC-"
 _AUTH_HEADER_REGEX = re.compile(
     r"^HMAC-SHA256 SignedHeaders=(?P<headers>[\w;-]*)&Signature=(?P<signature>[^\W_]*$)"
 )
+TIMESTAMP_HEADER = "X-Authorization-Timestamp"
+CONTENT_HASH_HEADER = "X-Content-SHA256"
 
 
-def _sign_message(message: str, hmac_key: str = None) -> str:
-    key = hmac_key or settings.DEFAULT_HMAC_KEY
-    encoded_message = base64.b64encode(message.encode())
-    return hmac.new(
-        key=key.encode(), digestmod=hashlib.sha256, msg=encoded_message
-    ).hexdigest()
-
-
-def _get_signed_components(
-    request_url: str, request_method: str, body: str | bytes
-) -> tuple[list[str], dict[str, str]]:
-    parsed_url = urllib.parse.urlparse(request_url)
-    if isinstance(body, str):
-        body = body.encode()
-    content_hash = hashlib.sha256(body).hexdigest if body else None
-    auth_timestamp = datetime.now(UTC)
-    signed_segments = [
-        request_method,
-        parsed_url.path,
-        parsed_url.query,
-        str(auth_timestamp),
-        content_hash,
-    ]
-    # Filter out query string and content_hash if none present
-    signed_segments = [segment for segment in signed_segments if segment]
-    signed_headers = {"X-Authorization-Timestamp": auth_timestamp}
-    if content_hash:
-        signed_headers["X-Content-SHA256"] = content_hash
-    return signed_segments, signed_headers
+###
+# public
 
 
 def make_signed_headers(
```

Comment:
It seems that this method is only invoked in tests to mock an HMAC-authenticated requests from OSF. If so we should probably make a comment, unless there are plans to have GV make HMAC requests to other services.

Reply:
i don't think we have specific plans for GV to send hmac-verified requests to other services, but i think having this function here seems better than moving it to a test helper -- more easily allows possible futures where gv does that or where we consolidate hmac utils as part of a shared dependency and use the same code on both ends of the hmac requests